### PR TITLE
[RFC] add find and replace and command-palette, fix multi cell selection for 4.1

### PR DESCRIPTION
**Pull request by jwkvam**
I added the find and replace shortcut and fixed the multicell selection for myself.

I don't know if this works for everyone, particularly the split between the `jupyter-notebook` commands versus the `ipython` commands so I don't know if you want to merge, but I thought I'd put it here in case someone finds it useful.

I added the `command-palette` shortcut as well since it was easy enough to do so.

### CODE REVIEW

File: vim_binding.js [hunk 1]
```diff
@@ -306,8 +308,8 @@ define([
       'z,t': 'jupyter-notebook:scroll-cell-top',
       'ctrl-j': 'jupyter-notebook:select-next-cell',
       'ctrl-k': 'jupyter-notebook:select-previous-cell',
-      'shift-j': 'jupyter-notebook:extend-marked-cells-below',
-      'shift-k': 'jupyter-notebook:extend-marked-cells-above',
+      'shift-j': 'jupyter-notebook:extend-selection-below',
+      'shift-k': 'jupyter-notebook:extend-selection-above',
```
Comment (lambdalisue) [inline vim_binding.js:312]: Note: They change the name again....
https://github.com/jupyter/notebook/commit/f05dfae0f3b2368bead6e374779ac21f7948e245#diff-52f8e2b09a62fa92d0a7142010ec4a7bL182

`extend-marked-cells-below` does not exist in master, 4.x, 4.0.x thus it should be `extend-selection-below`

File: vim_binding.js [hunk 2]
```diff
@@ -390,6 +392,7 @@ define([
     km.command_shortcuts.clear_shortcuts();
     addShortcuts(km.command_shortcuts, {
       'ctrl-c': 'ipython.interrupt-kernel',
+      'cmdtrl-shift-p': 'ipython.show-command-palette',
```
Comment (lambdalisue) [inline vim_binding.js:395]: Note: Exist in 4.x but 4.0.x https://github.com/jupyter/notebook/blob/4.x/notebook/static/notebook/js/actions.js#L420

File: vim_binding.js [hunk 3]
```diff
@@ -399,6 +402,7 @@ define([
       'esc': 'ipython.close-pager',
       'q': 'ipython.close-pager',
       'enter': 'ipython.enter-edit-mode',
+      'f': 'ipython.find-and-replace',
```
Comment (lambdalisue) [inline vim_binding.js:405]: [SHOULD]
It does not exist. Please create an alternative action for IPython 4.x

File: vim_binding.js [hunk 4]
```diff
@@ -406,8 +410,8 @@ define([
       'z,t': 'ipython.scroll-cell-top',
       'ctrl-j': 'ipython.select-next-cell',
       'ctrl-k': 'ipython.select-previous-cell',
-      'shift-j': 'ipython.extend-selection-next',
-      'shift-k': 'ipython.extend-selection-previous',
+      'shift-j': 'ipython.extend-selection-below',
```
Comment (lambdalisue) [inline vim_binding.js:413]: Note: Correct in 4.x https://github.com/jupyter/notebook/blob/4.x/notebook/static/notebook/js/actions.js#L420 but in 4.0.x https://github.com/jupyter/notebook/blob/4.0.x/notebook/static/notebook/js/actions.js#L114

**Comment by lambdalisue** [comment]
I need to settle the Jupyter version to support.
Does anyone know about the detail of Jupyter's versioning?

**Comment by jwkvam** [comment]
Thanks for the feedback, do you want to keep support for 4.0.x?

**Comment by lambdalisue** [comment]
Merged.

**Comment by lambdalisue** [comment]
I just ends up supporting 4.0.x so LGTM

**Pull request merged**